controllers/machinedeployment_sync.go

```diff
@@ -368,6 +367,23 @@ func calculateStatus(allMSs []*clusterv1.MachineSet, newMS *clusterv1.MachineSet
 		UnavailableReplicas: unavailableReplicas,
 	}
 
+	if *deployment.Spec.Replicas == status.ReadyReplicas {
+		status.Phase = string(clusterv1.MachineDeploymentPhaseRunning)
+	}
+	if *deployment.Spec.Replicas != status.ReadyReplicas && *deployment.Spec.Replicas > status.ReadyReplicas {
```

Comment:
You can omit the equality check, can't you?

Reply:
yes, removed it